bin/changelog.js

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env node
+'use strict';
+
+const requestPromise = require('request-promise');
+const chalk = require( 'chalk' );
+const octokit = require( '@octokit/rest' )();
+const promptly = require('promptly');
+
+const headers = {
+	'Content-Type': 'application/json;charset=UTF-8',
+	'Authorization': `token ${process.env.GH_API_TOKEN}`,
+	'Accept': 'application/vnd.github.inertia-preview+json',
+	'User-Agent': 'request'
+};
+
+const columnIds = [];
+
+const getPullRequestType = labels => {
+	const typeLabel = labels.find( label => label.name.includes( '[Type]' ) );
+	if ( ! typeLabel ) {
+		return 'Dev';
+	}
+	return typeLabel.name.replace( '[Type] ', '' );
+};
+
+const getLabels = labels => {
+	return labels
+		.filter( label => ! /\[.*\]/.test( label.name ) )
+		.map( label => label.name )
+		.join( ', ' );
+
+};
+const isCollaborator = async ( username ) => {
+	return requestPromise( {
+		url: `https://api.github.com/orgs/woocommerce/members/${ username }`,
+		headers,
+		resolveWithFullResponse: true
+	} ).then( response => {
+		return response.statusCode === 204;
+	} )
+		.catch( err => {
+			if ( err.statusCode !== 404 ) {
+				console.log( '🤯' );
+				console.log( err.message );
+			}
+		});
+}
+
+const writeEntry = async ( content_url ) => {
+	const options = {
+		url: content_url,
+		headers,
+		json: true
+	}
+	return requestPromise( options )
+		.then( async data => {
+			if ( data.pull_request ) {
+				const collaborator = await isCollaborator( data.user.login );
+				const type = getPullRequestType( data.labels );
+				const labels = getLabels( data.labels );
+				const labelTag = labels.length ? `(${ labels })` : '';
+				const authorTag = !! collaborator ? '' : `👏 @${ data.user.login }`;
+				const entry = `- ${ type }: ${ data.title } #${ data.number } ${ labelTag } ${ authorTag }`;
```

Comment:
Would it be possible to parse the PR for the suggested changelog entry in https://github.com/woocommerce/woocommerce-admin/blob/master/.github/PULL_REQUEST_TEMPLATE.md?

I was thinking logic along the lines of: look for text following `### Changelog Note:` that contains a prefix of `Tweak:|Dev:|Enhancement:...` and use that text if provided.

Reply:
It would be possible. I like using the title because (besides avoiding regex, which I ❤️) it encourages a short description. 

I'm open to it, however.

Reply:
Hmm, yeah here is a good example of a changelog note having a much different tone than the title: https://github.com/woocommerce/woocommerce-admin/pull/2423

How about this (in a folllow up): We change the template to prompt for an optional changelog entry and its required format and let the PR author know that otherwise the title will be used. Then we can scrape the description for a changelog entry in the changelog script.

Reply:
Sounds like a great way of handling it!